code/datums/gamemodes/game_mode.dm

```diff
@@ -499,25 +715,45 @@
 
 	return TRUE
 
+/datum/game_mode/proc/append_player_votes_link(mob/new_player/NP)
+	if(QDELETED(NP) || IsGuestKey(NP.key) || !SSdbcore.IsConnected())
+		return "" // append nothing
+
+	var/isadmin = check_rights(R_ADMIN, FALSE)
+	var/newpoll = FALSE
+	var/datum/DBQuery/query_get_new_polls = SSdbcore.NewQuery("SELECT id FROM [format_table_name("poll_question")] WHERE [(isadmin ? "" : "adminonly = false AND")] Now() BETWEEN starttime AND endtime AND id NOT IN (SELECT pollid FROM [format_table_name("poll_vote")] WHERE ckey = \"[sanitizeSQL(NP.ckey)]\") AND id NOT IN (SELECT pollid FROM [format_table_name("poll_textreply")] WHERE ckey = \"[sanitizeSQL(NP.ckey)]\")")
+	if(query_get_new_polls.Execute())
+		if(query_get_new_polls.NextRow())
+			newpoll = TRUE
+	qdel(query_get_new_polls)
 
-/datum/game_mode/proc/AttemptLateSpawn(mob/M, rank)
+	if(newpoll)
+		return "<p><b><a href='byond://?src=[REF(NP)];showpoll=1'>Show Player Polls</A> (NEW!)</b></p>"
+	else
+		return "<p><a href='byond://?src=[REF(NP)];showpoll=1'>Show Player Polls</A></p>"
+
+
+/datum/game_mode/proc/CanLateSpawn(mob/M, rank)
 	if(!isnewplayer(M))
-		return
+		return FALSE
 	var/mob/new_player/NP = M
-	if(!NP.IsJobAvailable(rank))
+	if(!NP.IsJobAvailable(rank, TRUE))
 		to_chat(usr, "<span class='warning'>Selected job is not available.<spawn>")
-		return
+		return FALSE
 	if(!SSticker || SSticker.current_state != GAME_STATE_PLAYING)
 		to_chat(usr, "<span class='warning'>The round is either not ready, or has already finished!<spawn>")
-		return
+		return FALSE
 	if(!GLOB.enter_allowed)
 		to_chat(usr, "<span class='warning'>Spawning currently disabled, please observe.<spawn>")
-		return
-
+		return FALSE
 	if(!SSjob.AssignRole(NP, rank, TRUE))
 		to_chat(usr, "<span class='warning'>Failed to assign selected role.<spawn>")
-		return
+		return FALSE
+
+	return TRUE
+
 
+/datum/game_mode/proc/AttemptLateSpawn(mob/new_player/NP, rank)
```

Comment:
This should rather be named "gamemode late spawn"

Reply:
Its already pretty clear
```
if(!SSticker?.mode.CanLateSpawn(src, rank))
	return
SSticker?.mode.AttemptLateSpawn(src, rank)
```

Reply:
Why can't the attempt_latespawn proc do it then?